Sources/MapboxDirections/Isochrones.swift

```diff
@@ -0,0 +1,170 @@
+    }
+    
+    /**
+     Begins asynchronously calculating isochrone contours using the given options and delivers the results to a closure.
+     
+     This method retrieves the contours asynchronously from the [Mapbox Isochrone API](https://docs.mapbox.com/api/navigation/isochrone/) over a network connection. If a connection error or server error occurs, details about the error are passed into the given completion handler in lieu of the contours.
+     
+     Contours may be displayed atop a [Mapbox map](https://www.mapbox.com/maps/).
+     
+     - parameter options: A `IsochroneOptions` object specifying the requirements for the resulting contours.
+     - parameter completionHandler: The closure (block) to call with the resulting contours. This closure is executed on the application’s main thread.
+     - returns: The data task used to perform the HTTP request. If, while waiting for the completion handler to execute, you no longer want the resulting contours, cancel this task.
+     */
+    @discardableResult open func calculate(_ options: IsochroneOptions, completionHandler: @escaping IsochroneCompletionHandler) -> URLSessionDataTask {
+        let session = (options: options, credentials: self.credentials)
+        let request = urlRequest(forCalculating: options)
+        let requestTask = urlSession.dataTask(with: request) { (possibleData, possibleResponse, possibleError) in
+            if let urlError = possibleError as? URLError {
+                DispatchQueue.main.async {
+                    completionHandler(session, .failure(.network(urlError)))
+                }
+                return
+            }
+            
+            guard let response = possibleResponse, ["application/json", "text/html"].contains(response.mimeType) else {
+                DispatchQueue.main.async {
+                    completionHandler(session, .failure(.invalidResponse(possibleResponse)))
+                }
+                return
+            }
+            
+            guard let data = possibleData else {
+                DispatchQueue.main.async {
+                    completionHandler(session, .failure(.noData))
+                }
+                return
+            }
+            
+            self.processingQueue.async {
+                do {
+                    let decoder = JSONDecoder()
+                    
+                    guard let disposition = try? decoder.decode(ResponseDisposition.self, from: data) else {
+                        let apiError = IsochroneError(code: nil, message: nil, response: possibleResponse, underlyingError: possibleError)
+
+                        DispatchQueue.main.async {
+                            completionHandler(session, .failure(apiError))
+                        }
+                        return
+                    }
+                    
+                    guard (disposition.code == nil && disposition.message == nil) || disposition.code == "Ok" else {
```

Comment:
There’s a bit of redundancy between this method and some of the `Directions` methods. As tail work, we could try factoring out some common tasks so it’s easier and less error-prone to spin up support for a new API.